src/app_service/service/community/service.nim

```diff
@@ -519,6 +519,13 @@ QtObject:
     # checking length is sufficient - communityTokensMetadata list can only extend
     return communityTokens.len != prevCommunityTokens.len
 
+  proc updateMemberRole(self: Service, community, prevCommunity: CommunityDto) =
```

Comment:
This procedure looks like O(M * N) a priori, where M is the number of chats and N is the number of members, however, I estimated that the actual time complexity is less that O(M * N) because when the signal is caught inside the controller, we do apply a filtering based on the chat. This makes the algorithm be ~O(N)